src/newlines.cpp

```diff
@@ -1962,7 +1962,7 @@ static void newline_before_return(chunk_t *start)
 
    chunk_t *pc = chunk_get_prev(nl);
    if (  !pc
-      || (pc->type == CT_BRACE_OPEN || pc->type == CT_VBRACE_OPEN))
+      || (pc->type == CT_BRACE_OPEN || pc->type == CT_VBRACE_OPEN || pc->type == CT_CASE_COLON))
```

Comment:
This only fixes the problem of `return` directly after a `:` So it doesn't really fix the problem.
Only the case of
```
    case "a":
        return foo;
```
will be fixed.

But if we add this to the test case(which we should) it still fails
```
    case "c":
        printf("hello world\n");
        return 0;
```
resulting in 
```
    case "c":
        printf("hello world\n");

        return 0;
```